src/owl.Router.js

```diff
@@ -34,40 +34,40 @@
                 return owl.Promise.resolve(null);
             }
 
-            return this.resolve(route).then(function (ret) {
-              return ret ? that.run(path, route) : null;
-            })
+            return this.resolve(route).then(function (resolveResult) {
+                const calledAllResolves = resolveResult.every(function (a) {return a;});
```

Comment:
This is not needed if promise resolved with false or 0 we need to just pass it to the controller.action, action will decide what to do.